File: problems/1679/solution.go
package main
func maxOperations(nums []int, k int) int {
    ans, m := 0, map[int]int{}
    
    for _, num := range nums {
        if _, ok := m[num]; ok && m[num] > 0 {
            m[num]--
            ans++
        } else {
            m[k - num]++
        }
    }
    
    return ans
}

File: problems/1679/solution.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        int ans = 0;
        unordered_map<int, int> m;
        
        for (int num: nums) {
            if (m.find(num) != m.end() && m[num] > 0) {
                m[num]--;
                ans++;
            } else {
                int t = k - num;
                m[t]++;
            }
        }
        
        return ans;
    }
};